admin/statistics/class-statistics-service.php

```diff
@@ -0,0 +1,172 @@
+	protected $statistics;
+
+	/**
+	 * WPSEO_Statistics_Service contructor.
+	 *
+	 * @param WPSEO_Statistics $statistics The statistics class to retrieve statistics from.
+	 */
+	public function __construct( WPSEO_Statistics $statistics = null ) {
+		if ( null === $statistics ) {
+			$statistics = new WPSEO_Statistics();
+		}
+
+		$this->statistics    = $statistics;
+	}
+
+	/**
+	 * Fetches statistics by REST request.
+	 *
+	 * @return WP_REST_Response The response object.
+	 */
+	public function get_statistics() {
+		$statistics = $this->statistic_items();
+
+		$header = __( 'Below are your published posts&#8217; SEO scores. Now is as good a time as any to start improving some of your posts!', 'wordpress-seo' );
+
+		$data = array(
+			'header'     => $header,
+			'seo_scores' => $statistics,
+		);
+
+		return new WP_REST_Response( $data );
+	}
+
+	/**
+	 * An array representing items to be added to the At a Glance dashboard widget
+	 *
+	 * @return array
+	 */
+	private function statistic_items() {
+		$transient = get_transient( self::CACHE_TRANSIENT_KEY );
+		$user_id   = get_current_user_id();
+
+		if ( isset( $transient[ $user_id ] ) ) {
+			return $transient[ $user_id ];
+		}
+
+		return $this->set_statistic_items_for_this_user( $transient );
+	}
+
+	/**
+	 * Set the cache for a specific user
+	 *
+	 * @param array|boolean $transient The current stored transient with the cached data.
+	 *
+	 * @return mixed
+	 */
+	private function set_statistic_items_for_this_user( $transient ) {
+		if ( $transient === false ) {
+			$transient = array();
+		}
+
+		$user_id               = get_current_user_id();
+		// Use array_values because array_filter may return non-zero indexed arrays.
+		$transient[ $user_id ] = array_values( array_filter( $this->get_seo_scores_with_post_count(), array( $this, 'filter_items' ) ) );
+
+		set_transient( self::CACHE_TRANSIENT_KEY, $transient, DAY_IN_SECONDS );
+
+		return $transient[ $user_id ];
+	}
+
+	/**
+	 * Get all SEO ranks and data associated with them.
+	 *
+	 * @return array An array of SEO scores and associated data.
+	 */
+	private function get_seo_scores_with_post_count() {
+		$ranks = WPSEO_Rank::get_all_ranks();
+
+		return array_map( array( $this, 'map_rank_to_widget' ), $ranks );
+	}
+
+	/**
+	 * Converts a rank to data usable in the dashboard widget.
+	 *
+	 * @param WPSEO_Rank $rank The rank to map.
+	 *
+	 * @return array The mapped rank.
+	 */
+	private function map_rank_to_widget( WPSEO_Rank $rank ) {
+		return array(
+			'seo_rank' => $rank->get_rank(),
+			'label'    => $this->get_label_for_rank( $rank ),
+			'count'    => $this->statistics->get_post_count( $rank ),
+			'link'     => $this->get_link_for_rank( $rank ),
+		);
+	}
+
+	/**
+	 * Returns a dashboard widget label to use for a certain rank.
+	 *
+	 * @param WPSEO_Rank $rank The rank to return a label for.
+	 *
+	 * @return string The label for the rank.
+	 */
+	private function get_label_for_rank( WPSEO_Rank $rank ) {
+		$labels = array(
```

Comment:
I am not sure about the working of the PHP internals, but I guess PHP will build this array every time this method is called.